src/MicrosoftAuthFlow.js

```diff
@@ -1,190 +1,199 @@
-
-        if (this.options.authTitle) {
-          const deviceToken = await this.xbl.getDeviceToken(this.options)
-          const titleToken = await this.xbl.getTitleToken(msaToken, deviceToken)
-          const xsts = await this.xbl.getXSTSToken(ut, deviceToken, titleToken)
-          return xsts
-        } else {
-          const xsts = await this.xbl.getXSTSToken(ut)
-          return xsts
-        }
-      }, () => { this.msa.forceRefresh = true }, 2)
-    }
-  }
-
-  async getMinecraftJavaToken (options = {}) {
-    assert(this.options.authTitle !== undefined, 'Please specify an "authTitle" in Authflow constructor')
-    const response = { token: '', entitlements: {}, profile: {} }
-    if (await this.mca.verifyTokens()) {
-      debug('[mc] Using existing tokens')
-      response.token = this.mca.getCachedAccessToken().token
-    } else {
-      this.xbl.relyingParty = Endpoints.PCXSTSRelyingParty
-      debug('[mc] Need to obtain tokens')
-      await retry(async () => {
-        const xsts = await this.getXboxToken()
-        debug('[xbl] xsts data', xsts)
-        response.token = await this.mca.getAccessToken(xsts)
-      }, () => { this.xbl.forceRefresh = true }, 2)
-    }
-
-    if (options.fetchEntitlements) {
-      response.entitlements = await this.mca.fetchEntitlements(response.token)
-    }
-    if (options.fetchProfile) {
-      response.profile = await this.mca.fetchProfile(response.token)
-    }
-
-    return response
-  }
-
-  async getMinecraftBedrockToken (publicKey) {
-    assert(this.options.authTitle !== undefined, 'Please specify an "authTitle" in Authflow constructor')
-    // TODO: Fix cache, in order to do cache we also need to cache the ECDH keys so disable it
-    // is this even a good idea to cache?
-    if (await this.mba.verifyTokens() && false) { // eslint-disable-line
-      debug('[mc] Using existing tokens')
-      return this.mba.getCachedAccessToken().chain
-    } else {
-      if (!publicKey) throw new Error('Need to specifiy a ECDH x509 URL encoded public key')
-      this.xbl.relyingParty = Endpoints.BedrockXSTSRelyingParty
-      debug('[mc] Need to obtain tokens')
-      return await retry(async () => {
-        const xsts = await this.getXboxToken()
-        debug('[xbl] xsts data', xsts)
-        const token = await this.mba.getAccessToken(publicKey, xsts)
-        // If we want to auth with a title ID, make sure there's a TitleID in the response
-        const body = JSON.parse(Buffer.from(token.chain[1].split('.')[1], 'base64').toString())
-        if (!body.extraData.titleId && this.options.authTitle) {
-          throw Error('missing titleId in response')
-        }
-        return token.chain
-      }, () => { this.xbl.forceRefresh = true }, 2)
-    }
-  }
-}
-
-function sha1 (data) {
-  return crypto.createHash('sha1').update(data ?? '', 'binary').digest('hex')
-}
-
-module.exports = MicrosoftAuthFlow
+const fs = require('fs')

```

Comment:
avoid changing all lines ends, so we can have a diff

Reply:
that will also prevent you from getting conflicts so easily

Reply:
Yeah, that's my bad on that one, here's the change in that file for easy viewing, just one added statement to the getXboxToken function.

https://github.com/PrismarineJS/prismarine-auth/pull/24/files#diff-466421b5023aca5be930859ce1246f70fb12b4abb1cfce0a168cd235ec93aa3bR120

Reply:
can you add a new commit here to fix this ?

Reply:
Fixed